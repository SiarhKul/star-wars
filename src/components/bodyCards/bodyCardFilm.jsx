import React from "react";

export const BodyCardFilm = ({ card: { title, director, release_date } }) => {
	return (
		<div className="body">
			<div className="body__text">Name:{title}</div>
			<div className="body__text">Director:{director}</div>
			<div className="body__text">Release date:{release_date}</div>
		</div>
	);
};
