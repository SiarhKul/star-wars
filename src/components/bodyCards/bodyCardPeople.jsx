import React from "react";

export const BodyCardPeople = ({ card }) => (
		<div className="body">
			<div className="body__text">Name:{card.name}</div>
			<div className="body__text">Birth year:{card?.birth_year}</div>
			<div className="body__text">Gender:{card?.gender}</div>
		</div>
	);
