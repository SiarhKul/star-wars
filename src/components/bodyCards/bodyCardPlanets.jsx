import React from "react";

export const bodyCardPlanets = ({ card }) => {
	return (
		<div className="body">
			<div className="body__text">Name:{card.name}</div>
			<div className="body__text">Population:{card.population}</div>
		</div>
	);
};
