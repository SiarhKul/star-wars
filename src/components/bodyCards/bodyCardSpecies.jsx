import React from "react";

export const BodyCardSpecies = ({ card }) => (
		<div className="body">
			<div className="body__text">Name:{card.name}</div>
			<div className="body__text">Language:{card.language}</div>
		</div>
	);
