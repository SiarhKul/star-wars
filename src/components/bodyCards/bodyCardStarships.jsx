import React from "react";

export const BodyCardStarships = ({ card }) => {
	return (
		<div className="body">
			<div className="body__text">Name:{card.name}</div>
			<div className="body__text">Cost:{card.cost_in_credits}</div>
			<div className="body__text">Passangers:{card.passengers}</div>
		</div>
	);
};
