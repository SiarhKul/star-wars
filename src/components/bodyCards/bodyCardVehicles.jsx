import React from "react";

export const BodyCardVehicles = ({ card }) => (
		<div className="body">
			<div className="body__text">Name:{card?.name}</div>
			<div className="body__text">Model:{card?.model}</div>
			<div className="body__text">Crew:{card?.crew}</div>
		</div>
	);
