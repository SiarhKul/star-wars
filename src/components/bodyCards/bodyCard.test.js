import React from 'react';
import { BodyCardFilm } from './bodyCardFilm';
import { BodyCardPeople } from './bodyCardPeople';
import { BodyCardPlanets } from './bodyCardPlanets';
import { BodyCardSpecies } from './bodyCardSpecies';
import { BodyCardStarships } from './bodyCardStarships';
import { BodyCardVehicles } from './bodyCardVehicles';

describe('Test bodyCards', () => {
  describe('Test <BodyCardFilm/>', () => {
    it('render snapshot <BodyCardFilm/> ', () => {
      const mockedProps = {
        card: {
          title: 'A New Hope',
          director: 'George Lucas',
          release_date: '1977-05-25',
        },
      };
      const component = shallow(<BodyCardFilm {...mockedProps} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Test <BodyCardPeople/> ', () => {
    it('render snapshot <BodyCarPeople/>', () => {
      const mockedProps = {
        card: {
          name: 'Luke Skywalker',
          birth_year: '19BBY',
          gender: 'male',
        },
      };
      const component = shallow(<BodyCardPeople {...mockedProps} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Test <BodyCardPlanets/> ', () => {
    it('render snapshot <BodyCarPlanets/>', () => {
      const mockedProps = {
        card: {
          name: 'Tatooine',
          population: 200000,
        },
      };
      const component = shallow(<BodyCardPlanets {...mockedProps} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Test <BodyCardPSpecies/> ', () => {
    it('render snapshot <BodyCarPSpecies/>', () => {
      const mockedProps = {
        card: {
          name: 'Human',
          language: 'Galactic Basic',
        },
      };
      const component = shallow(<BodyCardSpecies {...mockedProps} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Test <BodyCardStarships/> ', () => {
    it('render snapshot <BodyCarPStarships/>', () => {
      const mockedProps = {
        card: {
          name: 'CR90',
          cost_in_credits: 3500000,
          passengers: 600,
        },
      };
      const component = shallow(<BodyCardStarships {...mockedProps} />);
      expect(component).toMatchSnapshot();
    });
  });

  describe('Test <BodyCardVehicle/> ', () => {
    it('render snapshot <BodyCarVehicle/>', () => {
      const mockedProps = {
        card: {
          name: 'Sand Crawleer',
          mode: 'Digger Crawleer',
          crew: 46,
        },
      };
      const component = shallow(<BodyCardVehicles {...mockedProps} />);
      expect(component).toMatchSnapshot();
    });
  });
});
