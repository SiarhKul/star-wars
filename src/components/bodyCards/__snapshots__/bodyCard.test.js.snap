// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test bodyCards Test <BodyCardFilm/> render snapshot <BodyCardFilm/>  1`] = `
<div
  className="body"
>
  <div
    className="body__text"
  >
    Name:
    A New Hope
  </div>
  <div
    className="body__text"
  >
    Director:
    George Lucas
  </div>
  <div
    className="body__text"
  >
    Release date:
    1977-05-25
  </div>
</div>
`;

exports[`Test bodyCards Test <BodyCardPSpecies/>  render snapshot <BodyCarPSpecies/> 1`] = `
<div
  className="body"
>
  <div
    className="body__text"
  >
    Name:
    Human
  </div>
  <div
    className="body__text"
  >
    Language:
    Galactic Basic
  </div>
</div>
`;

exports[`Test bodyCards Test <BodyCardPeople/>  render snapshot <BodyCarPeople/> 1`] = `
<div
  className="body"
>
  <div
    className="body__text"
  >
    Name:
    Luke Skywalker
  </div>
  <div
    className="body__text"
  >
    Birth year:
    19BBY
  </div>
  <div
    className="body__text"
  >
    Gender:
    male
  </div>
</div>
`;

exports[`Test bodyCards Test <BodyCardPlanets/>  render snapshot <BodyCarPlanets/> 1`] = `
<div
  className="body"
>
  <div
    className="body__text"
  >
    Name:
    Tatooine
  </div>
  <div
    className="body__text"
  >
    Population:
    200000
  </div>
</div>
`;

exports[`Test bodyCards Test <BodyCardStarships/>  render snapshot <BodyCarPStarships/> 1`] = `
<div
  className="body"
>
  <div
    className="body__text"
  >
    Name:
    CR90
  </div>
  <div
    className="body__text"
  >
    Cost:
    3500000
  </div>
  <div
    className="body__text"
  >
    Passangers:
    600
  </div>
</div>
`;

exports[`Test bodyCards Test <BodyCardVehicle/>  render snapshot <BodyCarVehicle/> 1`] = `
<div
  className="body"
>
  <div
    className="body__text"
  >
    Name:
    Sand Crawleer
  </div>
  <div
    className="body__text"
  >
    Model:
  </div>
  <div
    className="body__text"
  >
    Crew:
    46
  </div>
</div>
`;
